server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -331,6 +345,13 @@ protected void onServerException(HttpServerChannel channel, Exception e) {
     protected void serverAcceptedChannel(HttpChannel httpChannel) {
         boolean addedOnThisCall = httpChannels.add(httpChannel);
         assert addedOnThisCall : "Channel should only be added to http channel set once";
+        openChannels.incrementAndGet();
```

Comment:
Making this depend on the lifecycle and complete `allClientsClosedListener` in multiple places seems a bit ugly, could we use a `RefCounted` with one ref for the transport and one for each open channel?

Reply:
++ fair point I pushed https://github.com/elastic/elasticsearch/pull/76257/commits/59ad20a34143e602d8ea7b41b4298d3c16659017